### PR TITLE
Teach flutter_tools to populate PUB_ENVIRONMENT

### PR DESCRIPTION
**Pull request by kevmoo**
Will allow telemetry reporting on pub.dartlang.org once
flutter moves to 1.23.0-dev.10.0

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
@johnmccutchan :D

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 1]
```diff
@@ -70,6 +72,30 @@ Future<Null> pubGet({
...
+
+/// Returns the environment value that should be used when running pub.
+///
+/// Includes any existing environment variable, if one exists.
+String _getPubEnvironment() {
+  final List<String> values = <String>[];
+
+  final String existing = platform.environment[_pubEnvironmentKey];
+
+  if (existing != null && existing.isNotEmpty) {
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/dart/pub.dart]: ((existing != null) && ..

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 2]
```diff
@@ -70,6 +72,30 @@ Future<Null> pubGet({
 
 final RegExp _analyzerWarning = new RegExp(r'^! \w+ [^ ]+ from path \.\./\.\./bin/cache/dart-sdk/lib/\w+$');
 
+/// The console environment key used by the pub tool.
+const String _pubEnvironmentKey = 'PUB_ENVIRONMENT';
+
+/// Returns the environment value that should be used when running pub.
+///
+/// Includes any existing environment variable, if one exists.
+String _getPubEnvironment() {
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/dart/pub.dart]: s/_getPubEnvironment/_getPubEnvironmentValue

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 3]
```diff
@@ -70,6 +72,30 @@ Future<Null> pubGet({
...
+  final List<String> values = <String>[];
+
+  final String existing = platform.environment[_pubEnvironmentKey];
+
+  if (existing != null && existing.isNotEmpty) {
+    values.add(existing);
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/dart/pub.dart]: Does the existing values string need to be split around ':' and then rejoined below?

**Review by johnmccutchan — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 2 above)
Comment (kevmoo) [inline packages/flutter_tools/lib/src/dart/pub.dart]: ack

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 1 above)
Comment (kevmoo) [inline packages/flutter_tools/lib/src/dart/pub.dart]: Really? ...uh...okay

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 3 above)
Comment (kevmoo) [inline packages/flutter_tools/lib/src/dart/pub.dart]: "need"? We could, but it feels unnecessary. Would it change behavior?

**Comment by kevmoo** [comment]
@johnmccutchan PTAL

**Review by johnmccutchan — APPROVED** [review]

**Comment by johnmccutchan** [comment]
LGTM

**Comment by kevmoo** [comment]
@johnmccutchan please do the honors – I lack the power! 😄

**Comment by eseidelGoogle** [comment]
I fixed your powers.  Thanks for the patch!

**Pull request merged**